src/auth/digest/eDirectory/ldap_backend.cc

```diff
@@ -213,8 +215,10 @@ getpassword(char *login, char *realm)
         if (usersearchfilter) {
             char escaped_login[1024];
             snprintf(searchbase, sizeof(searchbase), "%s", userbasedn);
+            searchbase[sizeof(searchbase)] = '\0';
             ldap_escape_value(escaped_login, sizeof(escaped_login), login);
             snprintf(filter, sizeof(filter), usersearchfilter, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login);
+            filter[sizeof(filter)] = '\0';
```

Comment:
Both changes above represent an out-of-bounds write AFAICT. Please fix.

How about using [std::snprintf](https://en.cppreference.com/w/cpp/io/c/fprintf) that the C++ standard guarantees to terminate?

```suggestion
            std::snprintf(searchbase, sizeof(searchbase), "%s", userbasedn);
            ldap_escape_value(escaped_login, sizeof(escaped_login), login);
            std::snprintf(filter, sizeof(filter), usersearchfilter, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login, escaped_login);
```

... and `#include <cstdio>`.